Traceback (most recent call last):
  File "/Users/jordanbarker/miniconda3/envs/py311/lib/python3.11/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/Users/jordanbarker/miniconda3/envs/py311/lib/python3.11/site-packages/nbclient/client.py", line 1305, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jordanbarker/miniconda3/envs/py311/lib/python3.11/site-packages/jupyter_core/utils/__init__.py", line 173, in wrapped
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jordanbarker/miniconda3/envs/py311/lib/python3.11/asyncio/base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/Users/jordanbarker/miniconda3/envs/py311/lib/python3.11/site-packages/nbclient/client.py", line 705, in async_execute
    await self.async_execute_cell(
  File "/Users/jordanbarker/miniconda3/envs/py311/lib/python3.11/site-packages/nbclient/client.py", line 1058, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/Users/jordanbarker/miniconda3/envs/py311/lib/python3.11/site-packages/nbclient/client.py", line 914, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
df = (pd.DataFrame(np.random.rand(10, 6),
                  columns = list('ABCDEF')) - 0.5).cumsum()
df['Group'] = [np.random.choice(list('abcd')) for _ in range(len(df))]
df['Date'] = pd.date_range('1/1/2017', periods=len(df), freq='D')
display(df)
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mNameError[0m                                 Traceback (most recent call last)
Cell [0;32mIn[35], line 1[0m
[0;32m----> 1[0m df [38;5;241m=[39m (pd[38;5;241m.[39mDataFrame([43mnp[49m[38;5;241m.[39mrandom[38;5;241m.[39mrand([38;5;241m10[39m, [38;5;241m6[39m),
[1;32m      2[0m                   columns [38;5;241m=[39m [38;5;28mlist[39m([38;5;124m'[39m[38;5;124mABCDEF[39m[38;5;124m'[39m)) [38;5;241m-[39m [38;5;241m0.5[39m)[38;5;241m.[39mcumsum()
[1;32m      3[0m df[[38;5;124m'[39m[38;5;124mGroup[39m[38;5;124m'[39m] [38;5;241m=[39m [np[38;5;241m.[39mrandom[38;5;241m.[39mchoice([38;5;28mlist[39m([38;5;124m'[39m[38;5;124mabcd[39m[38;5;124m'[39m)) [38;5;28;01mfor[39;00m _ [38;5;129;01min[39;00m [38;5;28mrange[39m([38;5;28mlen[39m(df))]
[1;32m      4[0m df[[38;5;124m'[39m[38;5;124mDate[39m[38;5;124m'[39m] [38;5;241m=[39m pd[38;5;241m.[39mdate_range([38;5;124m'[39m[38;5;124m1/1/2017[39m[38;5;124m'[39m, periods[38;5;241m=[39m[38;5;28mlen[39m(df), freq[38;5;241m=[39m[38;5;124m'[39m[38;5;124mD[39m[38;5;124m'[39m)

[0;31mNameError[0m: name 'np' is not defined

